array-manipulation-3/src/equal.js

```diff
@@ -0,0 +1,11 @@
+/* exported equal */
+function equal(first, second) {
+  for (let i = 0; i < first.length; i++) {
+    if (first[i] !== second[i]) {
+      return false;
+    } else if (first.length !== second.length) {
+      return false;
+    }
```

Comment:
It doesn't make sense to do this check over and over. Try moving the length comparison to before the loop and return early (a guard).